
<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <title>News - TDC</title>
    <link rel="stylesheet" href="/assets/css/app.css">
    <link rel="shortcut icon" type="image/png"
           href="/favicon.png" 
    />
    <script defer src="https://use.fontawesome.com/releases/v5.3.1/js/all.js"></script>
    <!-- Begin Jekyll SEO tag v2.8.0 -->
<title>News | TDC</title>
<meta name="generator" content="Jekyll v3.9.3" />
<meta property="og:title" content="News" />
<meta property="og:locale" content="en_US" />
<meta name="description" content="Artificial intelligence foundation for therapeutic science" />
<meta property="og:description" content="Artificial intelligence foundation for therapeutic science" />
<link rel="canonical" href="http://localhost:4000/news/" />
<meta property="og:url" content="http://localhost:4000/news/" />
<meta property="og:site_name" content="TDC" />
<meta property="og:type" content="website" />
<meta name="twitter:card" content="summary" />
<meta property="twitter:title" content="News" />
<script type="application/ld+json">
{"@context":"https://schema.org","@type":"WebPage","description":"Artificial intelligence foundation for therapeutic science","headline":"News","url":"http://localhost:4000/news/"}</script>
<!-- End Jekyll SEO tag -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-183522810-1"></script>
<script>
  window['ga-disable-UA-183522810-1'] = window.doNotTrack === "1" || navigator.doNotTrack === "1" || navigator.doNotTrack === "yes" || navigator.msDoNotTrack === "1";
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());
  gtag('config', 'UA-183522810-1');
</script><!-- head scripts --></head>

  <body>
    
<nav class="navbar is-primary" >
    <div class="container">
        <div class="navbar-brand">
            <a class="navbar-brand" href="/">
            <span><img src="/logonav.png" alt="Logo" style="height: auto; width: auto; max-height: 45px; max-width: 250px;"></span>
            </a>
            <a role="button" class="navbar-burger burger" aria-label="menu" aria-expanded="false" data-target="navMenu">
                <span aria-hidden="true"></span>
                <span aria-hidden="true"></span>
                <span aria-hidden="true"></span>
            </a>
        </div>
        <div class="navbar-menu" id="navMenu">
            <div class="navbar-end">
                
                
                    
                    <a href="/" class="navbar-item ">Home</a>
                    
                
                    
                    <a href="/start/" class="navbar-item ">Start</a>
                    
                
                    
                    <div class="navbar-item has-dropdown is-hoverable">
                        <a href="/overview/" class="navbar-link ">Datasets</a>
                        <div class="navbar-dropdown">
                            
                            <a href="/overview/" class="navbar-item ">Overview</a>
                            
                            <a href="/single_pred_tasks/overview" class="navbar-item ">Single-instance Prediction</a>
                            
                            <a href="/multi_pred_tasks/overview" class="navbar-item ">Multi-instance Prediction</a>
                            
                            <a href="/generation_tasks/overview" class="navbar-item ">Generation</a>
                            
                        </div>
                    </div>
                    
                
                    
                    <div class="navbar-item has-dropdown is-hoverable">
                        <a href="/fct_overview/" class="navbar-link ">Data Functions</a>
                        <div class="navbar-dropdown">
                            
                            <a href="/fct_overview/" class="navbar-item ">Overview</a>
                            
                            <a href="/functions/data_evaluation/" class="navbar-item ">Model Evaluation</a>
                            
                            <a href="/functions/data_split/" class="navbar-item ">Dataset Splits</a>
                            
                            <a href="/functions/data_process/" class="navbar-item ">Data Processing</a>
                            
                            <a href="/functions/oracles/" class="navbar-item ">Molecule Generation Oracles</a>
                            
                        </div>
                    </div>
                    
                
                    
                    <div class="navbar-item has-dropdown is-hoverable">
                        <a href="/benchmark/overview/" class="navbar-link ">Leaderboards</a>
                        <div class="navbar-dropdown">
                            
                            <a href="/benchmark/overview/" class="navbar-item ">Guidelines</a>
                            
                            <a href="/benchmark/admet_group/overview" class="navbar-item ">ADMET Group</a>
                            
                            <a href="/benchmark/drugcombo_group/overview" class="navbar-item ">DrugCombo Group</a>
                            
                            <a href="/benchmark/docking_group/overview" class="navbar-item ">Docking Group</a>
                            
                            <a href="/benchmark/dti_dg_group/overview" class="navbar-item ">DTI DG Group</a>
                            
                            <a href="/benchmark/scdti_group/overview" class="navbar-item ">Single-cell DTI Group</a>
                            
                            <a href="/benchmark/proteinpeptide_group/overview" class="navbar-item ">Protein-Peptide Binding Affinity Group</a>
                            
                            <a href="/benchmark/counterfactual_group/overview" class="navbar-item ">Counterfactual Prediction Group</a>
                            
                            <a href="/benchmark/clinical_trial/overview" class="navbar-item ">Clinical Trial Outcome Prediction Group</a>
                            
                        </div>
                    </div>
                    
                
                    
                    <a href="https://huggingface.co/tdc" class="navbar-item ">HF Models</a>
                    
                
                    
                    <a href="/news/" class="navbar-item is-active">News</a>
                    
                
                    
                    <a href="/team/" class="navbar-item ">Team</a>
                    
                
                <a href="https://arxiv.org/abs/2102.09548" class="navbar-item">NeurIPS Paper</a>
                <a href="https://www.nature.com/articles/s41589-022-01131-2" class="navbar-item">Nat Chem Bio Paper</a>
                <a href="https://tdc.readthedocs.io" class="navbar-item">Docs</a>
                <a href="https://github.com/mims-harvard/TDC" class="navbar-item">GitHub</a>
                
            </div>

        </div>
    </div>
</nav>

    
    


    <section class="section">
        <div class="container">
            <div class="columns">
                
                <div class="column is-12">
                    
                    
                    
                    
<div class="content">
    <h1 id="news">News</h1>

<hr />

  <p class="is-size-5"><strong>2024.06.22</strong> TDC-2 <a href="https://www.biorxiv.org/content/10.1101/2024.06.12.598655v2" a>preprint</a> is released!</p>
<h2>TDC-2 Features</h2>
    <ul>
        <li><strong>10+ new modalities</strong>: TDC-2 drastically expands the coverage of ML tasks across therapeutic pipelines and 10+ new modalities, spanning but not limited to single-cell gene expression data, clinical trial data, peptide sequence data, peptidomimetics protein-peptide interaction data regarding newly discovered ligands derived from AS-MS spectroscopy, novel 3D structural data for proteins, and cell-type-specific protein-protein interaction networks at single-cell resolution.</li>
        <li><strong>Single-cell atlases and foundation model embeddings</strong>: TDC-2 introduces over <strong>1,000</strong> multimodal datasets, spanning approximately 85 million cells and pre-calculated embeddings from 5 state-of-the-art single-cell models via <a href="https://cellxgene.cziscience.com/" a>CZ CELLxGENE Census</a> and the <a href="https://huggingface.co/tdc">TDC Model Hub</a></li>
        <li><strong>API-First Multimodal Retrieval API via TDC-2 MVC and Resource</strong>: TDC-2 drastically expands dataset retrieval capabilities available in TDC-1 beyond those of other leading benchmarks. The software architecture of TDC-2 was redesigned using the Model-View-Controller (MVC) design pattern.
        The MVC pattern supports the integration of multiple data modalities by using data mappings and views. The MVC-enabled-multimodal retrieval API is powered by TDC-2’s Resource Model and a Domain-Specific-Language.
        <ul>
          <li><strong>TDC-2 Domain-Specific Language</strong>: TDC-2 develops an Application-Embedded Domain-Specific Data Definition Programming Language 
            that facilitates the integration of multiple modalities by generating data views from a mapping of various datasets and functions for transformations, 
            integration, and multimodal enhancements while maintaining a high level of abstraction for the Resource framework.</li>
          <li><strong>TDC-2 Resource Model</strong>: The Commons introduces a redesign of TDC-1’s dataset layer into a new data model dubbed 
            the TDC-2 resource, developed under the MVC paradigm to integrate multiple modalities into the API-first model of TDC-2. We leverage 
            the CZ CellXGene to develop a TDC-2 Resource Model for constructing large-scale single-cell datasets that map gene expression profiles 
            of individual cells across tissues, healthy, and disease states.</li>
            <li><strong>Biomedical Knowledge Graphs and External APIs</strong>: We have developed a framework for biomedical knowledge graphs to 
            enhance the multimodality of dataset retrieval via TDC-2’s Resource Model. Our system leverages PrimeKG to integrate 
            <strong>20 high-quality resources to describe 17,080 diseases with 4,050,249 relationships</strong>. Our framework also 
            extends to external APIs, with data views currently leveraging BioPython, for obtaining nucleotide sequence information for a given 
            non-coding RNA ID from NCBI, and The Uniprot Consortium’s RESTful GET API for obtaining amino acid sequences</li>
          <li><strong>TDC-2 Model Hub</strong>: In addition, we’ve developed a framework that allows access to predictive and foundation 
            embedding models under diverse biological contexts via the TDC-2 Model Hub. TDC-2 releases AI-powered endpoints via 
            The Commons' Model Hub, which enhances multimodal retrieval capabilities by providing access to protein embeddings 
            under cell-type-specific biological contexts and model predictions for key biomedical challenges including but not limited to 
            binary classification on SMILES strings for Ether-a-go-go-related gene blockers, blood-brain-barrier permeability, and CYP3A4 
            inhibition. These models can be fine-tuned under TDC-2's fine-tuning paradigm and/or used for innovative downstream tasks.</li>
        </ul></li>
        <li><strong>7 innovative multimodal ML Tasks and benchmarks</strong>: TDC-2 introduces 7 novel ML tasks with fine-grained biological contexts: 
          contextualized drug-target identification, single-cell chemical/genetic perturbation response prediction, protein-peptide binding 
          affinity prediction task, and clinical trial outcome prediction task, which introduce antigen-processing-pathway-specific, 
          cell-type-specific, peptide-specific, and patient-specific biological contexts. TDC-2 also releases benchmarks evaluating 
          15+ state-of-the-art models across 5+ new learning tasks evaluating models on diverse biological contexts and sampling approaches. 
          Among these, TDC-2 provides the first benchmark for context-specific learning. TDC-2, to our knowledge, is also the first to introduce 
          a protein-peptide binding interaction benchmark. TDC-2's tasks, frameworks, datasets, and models are tailored to take on some of the most 
          pressing machine learning challenges in biomedicine, including but not limited to cell-type-specific machine learning modeling and 
          evaluation, the inferential gap in precision medicine, negative-sampling challenges in peptidomimetics, and model generalizability 
          across unseen cell lines and perturbations.</li>
        <li><strong>Performance Optimization</strong>: Implement effective strategies to boost your code's performance, ensuring faster, more efficient applications.</li>
        <li><strong>Collaboration Enhancement</strong>: Foster better teamwork through improved code readability and maintainability, making collaborative projects smoother and more productive.</li>
    </ul>
  <p>For more information on these and additional features, please refer to the <a href="https://www.biorxiv.org/content/10.1101/2024.06.12.598655v2">bioRxiv preprint</a>.</p>
  <hr />

<p class="is-size-5"><strong>2023.07.10</strong> TDC <code>0.4.1</code> is released! TDC has a new exciting task on clinical trial outcome prediction  (Thanks to Tianfan)! Checkout <a href="/multi_pred_tasks/trialoutcome/">here</a> for more information.</p>

<hr />

<p class="is-size-5"><strong>2023.04. 17</strong> TDC <code>0.4.0</code> is released! We're excited to announce the release of a new interface <code>tdc_hf_interface</code> that allows users to easily access and leverage pre-trained models hosted at <a href="https://huggingface.co/tdc"> HuggingFace on TDC datasets and tasks</a>. In this first batch, we've released nine pre-trained models from <a href="https://github.com/kexinhuang12345/DeepPurpose">DeepPurpose</a> that cover three popular ADMET datasets in the Commons. To load our pre-trainend model, simply do the following:
 </p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="kn">from</span> <span class="nn">tdc</span> <span class="kn">import</span> <span class="n">tdc_hf_interface</span>
<span class="n">tdc_hf</span> <span class="o">=</span> <span class="n">tdc_hf_interface</span><span class="p">(</span><span class="s">"BBB_Martins-AttentiveFP"</span><span class="p">)</span>
<span class="n">dp_model</span> <span class="o">=</span> <span class="n">tdc_hf</span><span class="p">.</span><span class="n">load_deeppurpose</span><span class="p">(</span><span class="s">'./data'</span><span class="p">)</span>
<span class="n">tdc_hf</span><span class="p">.</span><span class="n">predict_deeppurpose</span><span class="p">(</span><span class="n">dp_model</span><span class="p">,</span> <span class="p">[</span><span class="s">'CC(=O)NC1=CC=C(O)C=C1'</span><span class="p">])</span>
</code></pre></div></div>

<p class="is-size-5">The TDC-HF space is located at <a href="https://huggingface.co/tdc">here</a>. Stay tuned for more exciting pre-trained models, tasks &amp; demos!</p>

<hr />

<p class="is-size-5"><strong>2023.01.26</strong> TDC <code>0.3.9</code> is released! Here are the changes: </p>

<ul class="is-medium">
    <li> TDC has 9 new datasets on high throughput screening <code>HTS</code>. These assays cover a wide range of protein target classes and are carefully collated through confirmation screens to validate active compounds. See <a href="/single_pred_tasks/hts/#butkiewicz-et-al"> here </a> on how to access them!</li>
    </ul>

<table>
  <thead>
    <tr>
      <th>Protein Target Class</th>
      <th>PubChem AID</th>
      <th>Protein Target</th>
      <th>Total # of Molecules</th>
      <th># of Active Molecules</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>GPCR</td>
      <td>435008</td>
      <td>Orexin1 Receptor</td>
      <td>218,158</td>
      <td>233</td>
    </tr>
    <tr>
      <td>GPCR</td>
      <td>1798</td>
      <td>M1 Muscarinic Receptor Agonists</td>
      <td>61,833</td>
      <td>187</td>
    </tr>
    <tr>
      <td>GPCR</td>
      <td>435034</td>
      <td>M1 Muscarinic Receptor Antagonists</td>
      <td>61,756</td>
      <td>362</td>
    </tr>
    <tr>
      <td>Ion Channel</td>
      <td>1843</td>
      <td>Potassium Ion Channel Kir2.1</td>
      <td>301,493</td>
      <td>172</td>
    </tr>
    <tr>
      <td>Ion Channel</td>
      <td>2258</td>
      <td>KCNQ2 Potassium Channel</td>
      <td>302,405</td>
      <td>213</td>
    </tr>
    <tr>
      <td>Ion Channel</td>
      <td>463087</td>
      <td>Cav3 T-type Calcium Channels</td>
      <td>100,875</td>
      <td>703</td>
    </tr>
    <tr>
      <td>Transporter</td>
      <td>488997</td>
      <td>Choline Transporter</td>
      <td>302,306</td>
      <td>252</td>
    </tr>
    <tr>
      <td>Kinase</td>
      <td>2689</td>
      <td>Serine/Threonine Kinase 33</td>
      <td>319,792</td>
      <td>172</td>
    </tr>
    <tr>
      <td>Enzyme</td>
      <td>485290</td>
      <td>Tyrosyl-DNA Phosphodiesterase</td>
      <td>341,365</td>
      <td>281</td>
    </tr>
  </tbody>
</table>

<ul class="is-medium">
    <li> TDC has an additional dataset on hERG in the <code>Tox</code> task. See <a href="/single_pred_tasks/tox/#herg-karim-et-al"> here </a> for more info!</li>
    <li> TDC now follows black code style!</li>
</ul>

<hr />

<p class="is-size-5"><strong>2022.11.03</strong> TDC <code>0.3.8</code> is released! Here are the changes: </p>

<ul class="is-medium">
    <li> TDC has a new task on structure-based drug design <code>SBDD</code> with four datasets PDBBind, DUD-E, scPDB. See <a href="/generation_tasks/sbdd/"> here </a> on how to access them!</li>
    <li> To support evaluation of SBDD tasks, we also include two evaluation metrics (RMSD, Kabsch-RMSD) that compare distances between two structures. See <a href="/functions/data_evaluation/#rmsd">here</a> for more info.</li>    
    <li> TDC has a new dataset on  PAMPA (parallel artificial membrane permeability assay), which is a commonly employed assay to evaluate drug permeability across the cellular membrane in the <code>ADME</code> task. See <a href="/single_pred_tasks/adme/#pampa-permeability-ncats"> here </a> for more info!</li>
</ul>

<hr />

<p class="is-size-5"><strong>2022.09.06</strong> TDC <code>0.3.7</code> is released! Here are the changes: </p>

<ul class="is-medium">
    <li> TDC has a new evaluation metric on logAUC. See <a href="/functions/data_evaluation/#range-logauc"> here </a> and the <a href="https://github.com/mims-harvard/TDC/pull/152">PR</a>.</li>
    <li> TDC now supports graphein protein 3D representation for antibody develop-ability prediction. See <a href="https://github.com/mims-harvard/TDC/blob/main/tutorials/graphein_demo_developability.ipynb">tutorial</a> and the <a href="https://github.com/mims-harvard/TDC/pull/159">PR</a>.</li>
    <li> <code>QM</code> task are now in 3D format. See <a href="https://github.com/mims-harvard/TDC/pull/160">here</a>.</li>
    <li> TDC has a harmonize function to deal with duplicated experimental entries in DTI. See <a href="https://github.com/mims-harvard/TDC/pull/162">here</a>.</li>
    <li> TDC now has a dataloader for PrimeKG as an auxilliary resource. See how to access PrimeKG <a href="/resources/overview/#precision-medicine-knowledge-graph-primekg">here</a>.</li>
    <li> TDC fixed static scikit-learn version issue for gsk3b, jnk3, drd2 oracles. See <a href="https://github.com/mims-harvard/TDC/pull/168">here</a> for more info.</li>
    <li> The PPBR dataset in ADME task now has additional species information and the default is now only containing homo sapiens while you can retrieve other species via a TDC function. See <a href="/single_pred_tasks/adme/#ppbr-plasma-protein-binding-rate-astrazeneca">here</a> for more info.</li>
    
</ul>

<hr />

<p class="is-size-5"><strong>2022.02.19</strong> TDC <code>0.3.6</code> is released! TDC has a new task on TCR-Epitope Binding prediction  (Thanks to Anna and Jannis)! Checkout <a href="/multi_pred_tasks/tcrepitope/">here</a> for more information.</p>

<hr />

<p class="is-size-5"><strong>2022.01.23</strong> TDC <code>0.3.5</code> is released! Here are the changes: </p>

<ul class="is-medium">
    <li> TDC has a new large hERG dataset in <code>Tox</code> (Thanks to <a href="https://github.com/benb111">Ben</a>)! Checkout <a href="/single_pred_tasks/tox/#herg-central">here</a> for more information.</li>
</ul>

<ul class="is-medium">
    <li> TDC has an updated ChEMBL library (Version 29) in <code>MolGen</code>! The previous version is also still kept available. Checkout <a href="/generation_tasks/molgen/#chembl">here</a> for more information.</li>
</ul>

<ul class="is-medium">
    <li> Reaction type information can be found within split by turning on the include_reaction_type flag for USPTO-50 in <code>RetroSyn</code>!  Checkout <a href="/generation_tasks/retrosyn/#uspto-50k">here</a> for more information.</li>
</ul>

<ul class="is-medium">
    <li> Fixed bug on cold split for higher order (&gt;2) multi-instance prediction tasks! (Thanks to <a href="https://github.com/ZOE-V">Zoe</a> !)  Checkout <a href="https://github.com/mims-harvard/TDC/issues/136">here</a> for more information.</li>
</ul>

<hr />

<p class="is-size-5"><strong>2021.12.28</strong> TDC <code>0.3.4</code> is released! Bug fixes on docking oracles and KL divergence measure. </p>

<hr />

<hr />

<p class="is-size-5"><strong>2021.11.25</strong> TDC <code>0.3.3</code> is released! Now added extended support for cold split in multi prediction tasks, see <a href="https://github.com/mims-harvard/TDC/pull/127">this issue</a>! </p>

<hr />

<p class="is-size-5"><strong>2021.10.17</strong> TDC <code>0.3.2</code> is released! We have added support for harmonizing same DTIs with different affinities (KIBA, DAVIS Updated accordingly, see <a href="https://github.com/mims-harvard/TDC/issues/98">this issue</a>); support for label name retrieval for TWOSIDES (<a href="https://github.com/mims-harvard/TDC/issues/121">this issue</a>), and add gene symbol info to GDSC (<a href="https://github.com/mims-harvard/TDC/issues/122">this issue</a>).</p>

<hr />

<p class="is-size-5"><strong>2021.09.04</strong> TDC <code>0.3.0</code> is released! We have greatly restructured the code to be contributor friendly while keeping most interfaces the same. We also release the documentation for TDC package at <a href="https://tdc.readthedocs.io/">here</a>.</p>
<hr />

<p class="is-size-5"><strong>2021.05.30</strong> TDC updates to <code>0.2.0</code>, major changes:</p>

<ul class="is-medium">
    <li> TDC has a new molecule generation benchmark on docking scores! Checkout <a href="/benchmark/docking_group/overview">here</a> for more information.</li>
</ul>

<hr />

<p class="is-size-5"><strong>2021.03.24</strong> TDC updates to <code>0.1.9</code>, major changes:</p>

<ul class="is-medium">
    <li> TDC now supports molecule filters! Checkout <a href="/functions/data_process/#molecule-filters">here</a> for more information.</li>
</ul>

<hr />

<p class="is-size-5"><strong>2021.03.17</strong> TDC updates to <code>0.1.8</code>, major changes:</p>

<ul class="is-medium">
    <li> Leaderboard is reformulated and we invite submission for each individual benchmark! Checkout <a href="/benchmark/overview">here</a> for more information.</li>
</ul>

<hr />

<p class="is-size-5"><strong>2021.02.26</strong> TDC updates to <code>0.1.7</code>, major changes:</p>

<ul class="is-medium">
    <li> Streamlined leaderboard programming framework! Checkout <a href="/benchmark/overview">here</a> for more information.</li>
    <li> Label log transformation supported. Checkout <a href="/functions/data_process/#label-units-conversion">here</a> for more information.</li>
</ul>

<hr />

<p class="is-size-5"><strong>2021.02.18</strong> TDC just released the white paper in arXiv! Here is the <a href="https://arxiv.org/abs/2102.09548">link</a> to the paper.

<hr />

<p class="is-size-5"><strong>2021.02.04</strong> TDC updates to <code>0.1.6</code>, major changes:</p>

<ul class="is-medium">
    <li><strong> New Leaderboard! </strong> Just released the second leaderboard on drug combination response prediction! Checkout <a href="/benchmark/drugcombo_group">here</a> for usage.</li>
</ul>

<hr />

<p class="is-size-5"><strong>2021.01.16</strong> TDC updates to <code>0.1.5</code>, major changes:</p>

<ul class="is-medium">
    <li><strong> New Oracles! </strong> Added four realistic oracles from docking scores and synthetic accessibility scores! Checkout <a href="/functions/oracles">here</a> for usage.</li>
</ul>

<hr />

<p class="is-size-5"><strong>2021.01.09</strong> TDC updates to <code>0.1.4</code>, major changes:</p>

<ul class="is-medium">
    <li><strong> New Function! </strong> Added a data processing helper to map among ~15 molecular formats in 2 lines of code (For 2D: from SMILES/SEFLIES and convert to SELFIES/SMILES, Graph2D, PyG, DGL, ECFP2-6, MACCS, Daylight, RDKit2D, Morgan, PubChem; For 3D: from XYZ, SDF files to Graph3D, Columb Matrix). Checkout <a href="/functions/data_process/#molecule-conversion">here</a> for usage.</li>
    <li><strong> Quality Check! </strong> Canonicalize SMILES on DTI datasets with Drug, Target IDs added. Checkout <a href="/multi_pred_tasks/DTI/"><code>DTI</code></a>.</li>
</ul>

<hr />

<p class="is-size-5"><strong>2020.12.30</strong> TDC updates to <code>0.1.3</code>, major changes:</p>

<ul class="is-medium">
    <li><strong> New Dataset! </strong> Added a new therapeutic task CRISPR Repair Outcome Prediction! Checkout <a href="/single_pred_tasks/CRISPROutcome/"><code>CRISPROutcome</code></a>.</li>
    <li><strong> New Function! </strong> Added a data processing helper to map SMILES string to popular cheminformatics fingerprints (ECFP2, ECFP4, ECFP6, MACCS, Daylight-type, RDKit2D, Morgan, Pubchem)! Checkout <a href="/functions/data_process/#molecular-fingerprint-mapping">here</a> for usage.</li>
</ul>

<hr />

<p class="is-size-5"><strong>2020.12.24</strong> TDC updates to <code>0.1.2</code>, major changes:</p>

<ul class="is-medium">
    <li><strong> Leaderboard Release! </strong> TDC's first leaderboard on ADMET prediction is released. You can find the leaderboard guide <a href="/benchmark/overview">here</a>, where we provide a <code>BenchmarkGroup</code> class to do model building on leaderboard tasks rapidly. The ADMET leaderboard is <a href="/benchmark/admet_group/">here</a>.</li>
</ul>

<hr />

<p class="is-size-5"><strong>2020.12.19</strong> TDC updates to <code>0.1.1</code>, major changes:</p>

<ul class="is-medium">
    <li><strong> Quality Check and New datasets! </strong> We replaced VD, Half Life and Clearance datasets in <code>ADME</code> from new sources that have higher qualities. We also added LD50 to <code>Tox</code>.  </li>
</ul>

<hr />

<p class="is-size-5"><strong>2020.12.15</strong> TDC updates to <code>0.1.0</code>, major changes:</p>

<ul class="is-medium">
    <li><strong> Five New Datasets! </strong> Added CYP2C9/2D6/3A4 Substrate, for <code>ADME</code>, Carcinogens for <code>Tox</code> and NCI-60 for <code>DrugSyn</code>. </li>
    <li><strong>Quality Check.</strong> We conducted a canonicalization of all SMILES and removed ones that return errors in the <code>ADME</code>, <code>Tox</code>, and <code>HTS</code> datasets. </li>
</ul>

<hr />

<p class="is-size-5"><strong>2020.11.30</strong> TDC updates to <code>0.0.8</code>, major changes:</p>

<ul class="is-medium">
    <li><strong> Five New Datasets! </strong> Added hREG, DILI (Drug Induced Liver Injury), Skin Reaction, Ames Mutagenicity for <code>Tox</code> and PPBR from AstraZeneca for <code>ADME</code>. </li>
    <li><strong>Distribution Learning Metrics Moved to Evaluators.</strong> Checkout <a href="/functions/data_evaluation"> here </a> for the updated usage.</li>
    <li><strong>Meta Oracles.</strong> We included a helper function where you can specify your own set of molecules for Rediscovery, Similarity, Medians, Isomers. Checkout an example usage <a href="/functions/oracles/#rediscovery"> here</a>. </li>
    <li><strong>Tutorials.</strong> We have provided various tutorials for you to start using TDC. Click <a href="https://github.com/mims-harvard/TDC/tree/master/tutorials"> here </a>.
 </li>
</ul>

<hr />
</p>

</div>
                </div>
                
            </div>
        </div>
    </section>
    
        <footer class="footer">
    <div class="container">
        
        <div class="columns is-mobile">
            <div class="column is-8 has-text-left is-vcentered">
                <a class="navbar-brand" href="/">
                    <span><img src="/tdc_horizontal.png" alt="Logo" style="max-height: 40px; max-width: 250px;"></span>
                </a>
            </div>
            <div class="column is-4 has-text-right is-vcentered">
                <a href="https://arxiv.org/abs/2102.09548">
                    <span class="icon is-large">
                      <i class="fas fa-file-alt fa-3x"></i>
                    </span>
                </a>

                <a href="https://github.com/mims-harvard/TDC">
                    <span class="icon is-large">
                      <i class="fas fab fa-github fa-3x"></i>
                    </span>
                </a>

                <a href="https://twitter.com/ProjectTDC">
                    <span class="icon is-large">
                      <i class="fas fab fa-twitter fa-3x"></i>
                    </span>
                </a>

                <a href="https://join.slack.com/t/pytdc/shared_invite/zt-x0ujg5v6-zwtQZt83fhRdgrYjXRFz5g">
                    <span class="icon is-large">
                      <i class="fas fab fa-slack fa-3x"></i>
                    </span>
                </a>
            </div>
        </div>
        
    </div>
</footer>
    
    <script src="/assets/js/app.js" type="text/javascript"></script><!-- footer scripts --></body>
</html>

